lightning/src/chain/keysinterface.rs

```diff
@@ -40,6 +40,57 @@ use std::sync::atomic::{AtomicUsize, Ordering};
 use std::io::Error;
 use ln::msgs::DecodeError;
 
+/// Information about a spendable output to a P2WSH script. See
+/// SpendableOutputDescriptor::DynamicOutputP2WSH for more details on how to spend this.
+#[derive(Clone, Debug, PartialEq)]
+pub struct DynamicP2WSHOutputDescriptor {
+	/// The outpoint which is spendable
+	pub outpoint: OutPoint,
+	/// Per commitment point to derive delayed_payment_key by key holder
+	pub per_commitment_point: PublicKey,
+	/// The nSequence value which must be set in the spending input to satisfy the OP_CSV in
+	/// the witness_script.
+	pub to_self_delay: u16,
+	/// The output which is referenced by the given outpoint
+	pub output: TxOut,
+	/// The revocation_pubkey used to derive witnessScript
+	pub revocation_pubkey: PublicKey,
```

Comment:
No, this is the revocation point for the broadcast commitment transaction. I'll clarify the docs.

Reply:
Ah, we can't re-derive the `revocation_pubkey`, because that would require the counterparty's revocation base, which we don't have because we rederived it from scratch and therefore it doesn't have the counterparty's pubkeys.

However, we could still make it work if we looked up the populated `ChannelKeys` instead of re-deriving them.  So this brings up the question - why are we using `derive_channel_keys` instead of looking up the existing keys in some index?

Reply:
Right. Depending on the specific KeysInterface implementation we could look up a globally-cached remote public keys object. However we should try to be as implementation-agnostic as possible, so in this case I'm not sure it's ideal. We could provide the raw data to re-derive in the event, but I'm not sure what that protects against - if they even is bogus no funds loss can occur, the client may just create an invalid transaction.

Reply:
It's pretty minor, just felt a bit more "normalized" to provide just the minimum set of fields.